fishtest/fishtest/api.py

```diff
@@ -126,11 +126,12 @@ def stop_run(request):
   if not user or user['cpu_hours'] < 1000:
     return ''
 
-  run = request.rundb.get_run(request.json_body['run_id'])
-  run['stop_reason'] = request.json_body.get('message', 'No reason!')
-  request.actiondb.stop_run(username, run)
+  with request.rundb.active_run_lock(str(request.json_body['run_id'])):
+    run = request.rundb.get_run(request.json_body['run_id'])
+    run['stop_reason'] = request.json_body.get('message', 'No reason!')
```

Comment:
I would replace `'No reason!'` by `'api request'` or similar.

Reply:
Only the first stop_run is a bit expensive because all empty tasks are removed, the other calls are quite cheap when they are executed non parallel. Ignoring/skipping later stop_runs with additional code adds complexity with little gain. Note that  a stopped_run may also be rescheduled later...